src/umpire/ResourceManager.hpp

```diff
@@ -239,6 +248,10 @@ class ResourceManager {
     std::shared_ptr<strategy::AllocationStrategy>& findAllocatorForId(int id);
     std::shared_ptr<strategy::AllocationStrategy>& getAllocationStrategy(const std::string& name);
 
+
+    void registerAllocation(void* ptr, util::AllocationRecord* record);
```

Comment:
I believe that the implementation of ```registerAllocation()``` needs to confirm that its allocations are also not overlapping with external ones.

Reply:
I think that is a hopefully unlikely case, unless a user registers a pointer erroneously. I'd prefer to leave it unchecked, since it would add overhead to the registerAllocation call.

Reply:
Could we do something similar to what the DEBUG version of ```malloc()``` does?  Could we add the checks there, incur the overhead, and find the bug?

Reply:
Possibly. Do we want to establish a precedent for that? It would be hard to find overlapping externals without searching the whole AllocationMap.

Reply:
I think it could be done with one additional search.  If you looked for the ```ptr+size``` and examined the previous record to see if it found something in the range of ```[ptr, ptr+size]```, I think this would do it right?

Reply:
No, because there's nothing stopping a user registering an allocations in this situation (where the entries in this list are ordered by address, and each entry is the size):

Before: [ 2, 4, 2 ]
User registers addr with size 11: [_11_, 2, 4, 2]

The user could register an external allocation with any size.